### ISSUE TITLE
Invincible Mobs

### ISSUE THREAD

**Issue by EastTesla** [FTBTeam/FTB-Presents-Direwolf20-1.16#185]
* **Modpack Version**: 1.6.0

* **Issue**: Occasionally after loading into game spiders will spawn in that I can't kill. health get's drained to zero but they keep going

* **Is it Repeatable?**: I can't force it but it's happened multiple times.

**Comment by EastTesla** [comment]
Update: Its Happening with other mobs too now

**Comment by trip-dixon** [comment]
I'm having the same issue as well. It's happened with spiders, zombies, and skeletons (at least those are the ones I've noticed).

**Comment by PaulMuscat2407** [comment]
Same here, except it happens with all mobs. On load of the world ,the mobs take damage as normal but once they get to half a heart, there animations stop working and become invincible

**Comment by NerdySpider** [comment]
I can confirm this is happening with ALL mobs too... not sure what is causing this one chief!

**Comment by Sunekaer** [comment]
Hmm its not happening in my test world

**Comment by EastTesla** [comment]
Yeah it's happened with every mob for me atleast once, it lasts through relogs which is kinda annoying

**Comment by EastTesla** [comment]
I got a sreenshot where a lot of zombies had the problem, their pathfinding AI and animations seem to break once this happens. At the top it shows that the one im looking at doesn't have any health just so u know its not just the anim and pathfinding
![2021-02-04_16 56 42](https://user-images.githubusercontent.com/78547533/106927667-4c28ac80-670a-11eb-8ae4-33cd12ddce13.png)

**Comment by EastTesla** [comment]
They also are apparently immune to effects as they wont burn

![2021-02-04_17 00 14](https://user-images.githubusercontent.com/78547533/106927859-7e3a0e80-670a-11eb-8c72-9616d946733f.png)

**Comment by trip-dixon** [comment]
I had the pathing issues with other mobs as well. Some creepers will flash in a different pattern, and not explode. I didn't manage to get a screenshot of it... but now I don't know which creepers will explode on me and which ones won't.

Not what you want when one drops on you while in a desert temple... he turned out to be a dud

**Comment by EastTesla** [comment]
Just lost my hardcore world to a group of 6 invincible blazes. it's honestly pretty bad. any chance of there being a fix anytime soon?

**Comment by PaulMuscat2407** [comment]
I don't know if this helps, but I took a capture of a fresh world, walked around to find some mobs and it happened on every mob I punched

https://gyazo.com/15bf16b92508ef4eb277caacd163d76c

**Comment by PickaxeEngineer** [comment]
Can confirm this behaviour and - annoyingly - the inconsistency with it

**Comment by EastTesla** [comment]
I've done some messing around. what i first thought was that maybe a particular mod had put in a method of lowering lag and it just didnt work out well but i can go into a superflat reduce my render distance to two chunks and spawn in one entity and it'll still happen. then I gave up on trying to figure out why it was happening and tried figuring what was happening, just based off of pretty minimal messing around i think what's happening is a mod is making it so that sometimes the mods instead of dying are stopping being treated like entities which means even if you had cheats enabled it would be impossible to kill them. pretty surethe best way to deal with this currently is either with trapping them or mob cramming to try and get rid of the invincible ones if they're near your bed.

TL;DR - I don't think the invincible mobs are treated as enemies hence there's no way to kill them with or without cheats.

**Comment by SonicSpeedster16** [comment]
Totally spitballing here but it seems to have been determined that critical hits are somehow causing this. I noticed the following in the [change log for the version of Tetra](https://www.curseforge.com/minecraft/mc-mods/tetra/files/3187037) included in the 1.6.0 pack version:

* fixed thrown items not being able to do critical hits
* fixed crit block cheese

Has anyone maybe tried downgrading Tetra and seeing if the issue persists?

**Comment by Karo-DRS** [comment]
I am also having this issue. Almost 100% of the time crits will make the mob invincible. I'm about to try downgrading Tetra.

**Comment by Karo-DRS** [comment]
You can turn creepers into your friend.

![2021-02-04_15 55 16](https://user-images.githubusercontent.com/78567336/106960415-aa9d5d00-6701-11eb-8db9-bb5cf3937b9e.png)

**Comment by HU5N41N** [comment]
> 
> 
> I am also having this issue. Almost 100% of the time crits will make the mob invincible. I'm about to try downgrading Tetra.
> 
> Edit: Downgrading did not work. Bug still occurs.

Yeah i also tried removing tetra entirely that didn't work either lol, it started when i would crit something or anyone on the server and then i got crit which causes my screen to shake and become invincible as well as seems like nothing (mobs) acknowledges my existence the same thing is happening after a server restart as well

**Comment by Boyos1425** [comment]
So, It appears when I get critical hit it bugs out my game makes me invincible. I was able to replicate this with my son on the server. Logging doesn't fix it just stuck like your bugging out dyeing. Used fist, tetra and normal weapons. all when doing a crit will bug out the player or mob's. I was crit by a zombie and it bugged me out.

**Comment by VT-14** [comment]
After an hour long Binary Search, I found the problem mod to be MythicBotany-1.16.3-1.2.2.jar

The mod requires LibX and Botania, and by extension Curios and Patchouli.  Those mods without MythicBotany don't have the issue.  Adding MythicBotany causes mobs to become invincible when hit by a critical attack.

Updating to MythicBotany-1.16.3-1.2.3 did not fix the issue for me.

Removing MythicBotany does fix the issue on my end, though may not be desirable for players using the mod.

****

Mobs made invincible with the mod installed remain invincible even after it is removed.  Invincible Hostile mobs do despawn normally (closest player in same dimension >128 blocks away).  Other mobs (or players) will likely require using commands or out-of-game tools to fix.

**Comment by codedcosmos** [comment]
I experienced this glitch with enderman. But also myself. A friend killed me in game and I was unable to die properly. Typing /kill and re-logging didn't fix. Additionally restarting the server. I am perpetually at 0 hp, causing my screen to shake making it difficult to play.

**Comment by noeppi-noeppi** [comment]
The issue should be fixed in [MythicBotany-1.16.3-1.2.4](https://www.curseforge.com/minecraft/mc-mods/mythicbotany/files/3193542). All entities that are invincible because of this bug, will get fixed when they get damage the first time after updating MythicBotany.

**Comment by trip-dixon** [comment]
> The issue should be fixed in [MythicBotany-1.16.3-1.2.4](https://www.curseforge.com/minecraft/mc-mods/mythicbotany/files/3193542). All entities that are invincible because of this bug, will get fixed when they get damage the first time after updating MythicBotany.

Can confirm. I've never swapped out mods, or versions of mods before and this was very easy to do. Fixed the problem right away. Thanks @noeppi-noeppi !

**Comment by Sunekaer** [comment]
released 1.6.1
